serverless/aws/provider.py

```diff
@@ -150,8 +150,8 @@ def event_bridge(
 
         return fn
 
-    def s3(self, name, description, bucket, event, rules=None, handler=None, timeout=None, layers=None, **kwargs):
-        fn = S3Function(self.service, name, description, bucket, event, rules, handler, timeout, layers, **kwargs)
+    def s3(self, name, description, bucket, event, rules=None, existing=None, handler=None, timeout=None, layers=None, **kwargs):
```

Comment:
I agree, but in this case I think it made sense to add it after `rules` because they both apply to the S3 trigger of the function. Strictly speaking it's a breaking change, but I think a `minor` level version change will be enough.